invenio_circulation/permissions.py

```diff
@@ -45,6 +47,32 @@ def can(self):
     return type('Allow', (), {'can': can})()
 
 
+def has_loan_checkout_permission(*args, **kwargs):
```

Comment:
I thinks this logic should go to overlay as we have already the mechanism to pass your permission factories per transition.

Reply:
Yes, I was not sure if I should place it here, but if I won't, everyone will have a permission to checkout every loan, and I was not a fan of this idea, but if you guys think that's the way to go, then I will change it